src/scales/scale.time.js

```diff
@@ -334,27 +334,38 @@ function generate(min, max, minor, major, capacity, options) {
  * Returns the right and left offsets from edges in the form of {left, right}.
  * Offsets are added when the `offset` option is true.
  */
-function computeOffsets(table, ticks, min, max, options) {
+function computeOffsets(table, ts, min, max, options) {
 	var left = 0;
 	var right = 0;
-	var upper, lower;
-
-	if (options.offset && ticks.length) {
-		if (!options.time.min) {
-			upper = ticks.length > 1 ? ticks[1] : max;
-			lower = ticks[0];
-			left = (
-				interpolate(table, 'time', upper, 'pos') -
-				interpolate(table, 'time', lower, 'pos')
-			) / 2;
-		}
-		if (!options.time.max) {
-			upper = ticks[ticks.length - 1];
-			lower = ticks.length > 1 ? ticks[ticks.length - 2] : min;
-			right = (
-				interpolate(table, 'time', upper, 'pos') -
-				interpolate(table, 'time', lower, 'pos')
-			) / 2;
+	var timestamps = [];
+	var i, ilen, timestamp, length, upper, lower, divisor;
+
+	if (options.offset) {
+		[ts.labels, ts.data].forEach(function(data) {
+			if (!timestamps.length) {
```

Comment:
This is as intended. First, it processes `ts.labels` to pick up timestamps of labels, which is primarily used for offset calculation because bars are all on labels. But even if there are no labels, it should add offsets based on `ts.data`. That is the second path.

I will add a comment on it.